llvm/lib/IR/ConstantFPRange.cpp

```diff
@@ -112,7 +112,7 @@ ConstantFPRange ConstantFPRange::getNonNaN(const fltSemantics &Sem) {
 /// Return [-inf, V) or [-inf, V]
 static ConstantFPRange makeLessThan(APFloat V, FCmpInst::Predicate Pred) {
   const fltSemantics &Sem = V.getSemantics();
-  if (!(Pred & FCmpInst::FCMP_OEQ)) {
+  if (FCmpInst::isFalseWhenEqual(FCmpInst::getOrderedPredicate(Pred))) {
```

Comment:
I meant introduce a new local helper with the previous implementation, this still requires parsing through 2 steps

Reply:
Done.